hugegraph-core/src/main/java/com/baidu/hugegraph/auth/StandardAuthManager.java

```diff
@@ -516,10 +532,228 @@ public UserWithRole validateUser(String token) {
+        return commit(() -> {
+            // Create project admin group
+            if (project.adminGroupId() == null) {
+                HugeGroup adminGroup = new HugeGroup(project.adminGroupName());
+                adminGroup.creator(project.creator());
+                Id adminGroupId = this.createGroup(adminGroup);
+                project.adminGroupId(adminGroupId);
+            }
+
+            // Create project op group
+            if (project.opGroupId() == null) {
+                HugeGroup opGroup = new HugeGroup(project.opGroupName());
+                opGroup.creator(project.creator());
+                Id opGroupId = this.createGroup(opGroup);
+                project.opGroupId(opGroupId);
+            }
+
+            // Create project target to verify permission
+            HugeResource resource = new HugeResource(ResourceType.PROJECT,
+                                                     project.name(),
+                                                     null);
+            HugeTarget target = new HugeTarget(project.targetName(),
+                                               this.graph.name(),
+                                               "localhost:8080",
+                                               ImmutableList.of(resource));
+            target.creator(project.creator());
+            Id targetId = this.targets.add(target);
+            project.targetId(targetId);
+
+            Id adminGroupId = project.adminGroupId();
+            Id opGroupId = project.opGroupId();
+            HugeAccess adminGroupWriteAccess = new HugeAccess(
+                                                   adminGroupId, targetId,
+                                                   HugePermission.WRITE);
+            adminGroupWriteAccess.creator(project.creator());
+            HugeAccess adminGroupReadAccess = new HugeAccess(
+                                                  adminGroupId, targetId,
+                                                  HugePermission.READ);
+            adminGroupReadAccess.creator(project.creator());
+            HugeAccess opGroupReadAccess = new HugeAccess(opGroupId, targetId,
+                                                          HugePermission.READ);
+            opGroupReadAccess.creator(project.creator());
+            this.access.add(adminGroupWriteAccess);
+            this.access.add(adminGroupReadAccess);
+            this.access.add(opGroupReadAccess);
+            return this.project.add(project);
+        });
+    }
+
+    @Override
+    public HugeProject deleteProject(Id id) {
+        return this.commit(() -> {
+            LockUtil.Locks locks = new LockUtil.Locks(this.graph.name());
+            try {
+                locks.lockWrites(LockUtil.PROJECT_UPDATE, id);
+
+                HugeProject oldProject = this.project.get(id);
+                E.checkArgumentNotNull(oldProject,
+                                       "The project '%s' can't be found", id);
```

Comment:
improve it

Reply:
maybe change to "The project '%s' not found"?